naga/src/back/spv/writer.rs

```diff
@@ -971,6 +971,12 @@ impl Writer {
     ) -> Result<Word, Error> {
         let ty = &arena[handle];
         let id = if let Some(local) = make_local(&ty.inner) {
+            // If it's a type that needs SPIR-V capabilities, request them now.
+            // This needs to happen regardless of the LocalType lookup succeeding,
+            // because some types which map to the same LocalType have different
+            // capability requirements. See https://github.com/gfx-rs/wgpu/issues/5569
+            self.request_type_capabilities(&ty.inner)?;
+
```

Comment:
```suggestion
        // If it's a type that needs SPIR-V capabilities, request them now.
        // This needs to happen regardless of the LocalType lookup succeeding,
        // because some types which map to the same LocalType have different
        // capability requirements. See https://github.com/gfx-rs/wgpu/issues/5569
        self.request_type_capabilities(&ty.inner)?;
        let id = if let Some(local) = make_local(&ty.inner) {
```

I feel like we should take this out of the `if` as well in case we need to request capabilities for non local types.

Reply:
I can apply the suggestion and merge this if there are no concerns with it.